src/librustc_trans/partitioning.rs

```diff
@@ -587,7 +587,7 @@ fn characteristic_def_id_of_trans_item<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
             if let Some(impl_def_id) = tcx.impl_of_method(def_id) {
                 // This is a method within an inherent impl, find out what the
                 // self-type is:
-                let impl_self_ty = common::def_ty(tcx, impl_def_id, instance.substs);
+                let impl_self_ty = Instance::new(impl_def_id, instance.substs).ty(tcx);
```

Comment:
Same here as https://github.com/rust-lang/rust/pull/45525#discussion_r147662872.

Reply:
See https://github.com/rust-lang/rust/pull/45525#discussion_r152757123, still seems relevant to me (it's not a function or a global, but rather an `impl`'s `Self` type).